src/algorithms/synthesis/syrec_synthesis_additional_lines.cpp

```diff
@@ -0,0 +1,126 @@
+#include "algorithms/synthesis/syrec_synthesis_additional_lines.hpp"
+
+#include "core/utils/timer.hpp"
+
+namespace syrec {
+    bool SyrecSynthesisAdditionalLines::on_statement(const assign_statement& statement) {
+        std::vector<unsigned> lhs;
+        std::vector<unsigned> rhs;
+        get_variables(statement.lhs, lhs);
+        SyrecSynthesis::on_expression(statement.rhs, rhs, lhs, statement.op);
+
+        assert(lhs.size() == rhs.size());
+
+        bool status = false;
+
+        switch (statement.op) {
+            case assign_statement::add: {
+                status = increase(lhs, rhs);
+            } break;
+
+            case assign_statement::subtract: {
+                status = decrease(lhs, rhs);
+            } break;
+
+            case assign_statement::exor: {
+                status = bitwise_cnot(lhs, rhs);
+            } break;
+
+            default:
+                return false;
+        }
+        return status;
```

Comment:
while not as bad as for the `on_expression` function, there still is code duplication right here.
see the comment below for a possible suggestion.
if you don't feel comfortable refactoring that part, I could probably live with the current amount of code duplication.

Reply:
I tried reducing this as well. What do you suggest I do with the op_rhs_lhs_expression and expression_op_inverse function in the syrec_synthesis_additional_lines files?

Reply:
done 👍🏼